### PR TITLE
feat: team admin: see connected apps of team members

### LINKED ISSUE #8564: [CAL-1604] team admin: see connected apps of team members
CleanShot 2023-04-29 at 19 21 19@2x

with a tooltip of the connected calendar, i.e. "peer@cal.com"

remember, someone can have two google calendar

<sub>[CAL-1604](https://linear.app/calcom/issue/CAL-1604/team-admin-see-connected-apps-of-team-members)</sub>

### PR DESCRIPTION
**Pull request by abhijeetsingh-22**
## What does this PR do?
Added feature to show the connected apps of team members if the logged in user is Admin

Fixes #8564 
/claim #8564 

Following are the screenshots of team members page visible to different type of team users (admin, member, pending member)

Team Free Example User (Member)
Team Free Example User (Member)

Team Pro Example 2 User (Invitation Pending Member)
Team Pro Example 2 User (Pending Member)

Team Pro Example 3 User (Owner)
Team Pro Example 3 User (Owner)

Team Pro Example 3 User (Owner) video

[Team Pro Example 3 User (Owner) video.webm](https://github.com/calcom/cal.com/assets/55000107/67ec078d-86de-44c2-8219-eccccb9b0744)

## Type of change

- [ x] New feature (non-breaking change which adds functionality)

## How should this be tested?
To test this, create a team. Connect different apps from different member account. Now when you login as Admin you can view the logo of the connected apps of the team member. 
If the connected app is of type `calendar` then a tooltip will be shown on hovering on the logo which will show the email from which the calendar is connected

## Mandatory Tasks

- [ x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

**Review by PeerRich — DISMISSED** [review]
really awesome stuff! UI looks great, waiting for code review, maybe @alannnc can review this

File: packages/lib/server/queries/teams/index.ts [hunk 1]
```diff
@@ -111,16 +124,38 @@ export async function getTeamWithMembers(args: {
...
+
+          if (app?.categories.includes(AppCategories.calendar)) {
+            const externalId = connectedCalendars.find((cal) => cal.credentialId == id)?.primary?.email;
+            return { name: appMetaData?.name, logo: appMetaData?.logo, slug: appMetaData?.slug, externalId };
+          }
+          return { name: appMetaData?.name, logo: appMetaData?.logo, slug: appMetaData?.slug };
+        })
+        .sort((a, b) => (a.slug ?? "").localeCompare(b.slug ?? ""));
+      //Prevent credentials from leaking to frontend
+      const { credentials, destinationCalendar, selectedCalendars, ...rest } = {
```
Comment (alannnc) [inline packages/lib/server/queries/teams/index.ts:148]: Awesome job preventing credential leak

File: packages/lib/server/queries/teams/index.ts [hunk 2]
```diff
@@ -111,16 +124,38 @@ export async function getTeamWithMembers(args: {
...
+          const appMetaData = getAppFromSlug(app?.slug);
+
+          if (app?.categories.includes(AppCategories.calendar)) {
+            const externalId = connectedCalendars.find((cal) => cal.credentialId == id)?.primary?.email;
+            return { name: appMetaData?.name, logo: appMetaData?.logo, slug: appMetaData?.slug, externalId };
+          }
+          return { name: appMetaData?.name, logo: appMetaData?.logo, slug: appMetaData?.slug };
+        })
+        .sort((a, b) => (a.slug ?? "").localeCompare(b.slug ?? ""));
+      //Prevent credentials from leaking to frontend
```
Comment (alannnc) [inline packages/lib/server/queries/teams/index.ts:147]:
```suggestion
      // Prevent credentials from leaking to frontend
```

File: packages/lib/server/queries/teams/index.ts (hunk 1 above)
Comment (abhijeetsingh-22) [reply packages/lib/server/queries/teams/index.ts:148]: Thanks

**Review by alannnc — APPROVED** [review]
Really nice work! Pushed some minor changes.

**Pull request merged**